praw/objector.py

```diff
@@ -151,13 +155,17 @@ def objectify(
                 return data
             if "things" in data["json"]["data"]:  # Submission.reply
                 return self.objectify(data["json"]["data"]["things"])
+            if "rules" in data["json"]["data"]:
+                return self.objectify(loads(data["json"]["data"]["rules"]))
```

Comment:
Why is this recursing rather than just returning the direct result?

Reply:
Reddit gives us a JSON-encoded string as a value within a different JSON object (that we have already parsed). But only when creating rules (not when viewing them. It's really confusing. Something like

```
{"json": {"data": "{\"short_name\": \"my rule name\"}"}}
```

Reply:
If we parse that JSON then we know how to parse the resulting dict from a different case.

Reply:
The data is always in a list, even if there's one rule

Reply:
Not in this case. This one has to do with JSON parsing.